### PR TITLE
fix: full run with system v2 components

### PR DESCRIPTION
**Pull request by jsolaas**
## Why is this pull request needed?

ecalc model with system v2 did not run

## What does this pull request change?

- Set fuel on system v2
- Allow system v2 types in ConsumerSystemResult
- Calculate emissions for system v2

### CODE REVIEW

File: src/ecalc/libraries/libecalc/common/libecalc/core/ecalc.py [hunk 1]
```diff
@@ -72,6 +73,12 @@ def evaluate_energy_usage(self, variables_map: dto.VariablesMap) -> Dict[str, Ec
     def evaluate_emissions(
         self, variables_map: dto.VariablesMap, consumer_results: Dict[str, EcalcModelResult]
     ) -> Dict[str, Dict[str, EmissionResult]]:
+        """
+        Calculate emissions for fuel consumers and emitters
+        :param variables_map:
+        :param consumer_results:
+        :return:
```
Comment (jorgenengelsen) [inline src/ecalc/libraries/libecalc/common/libecalc/core/ecalc.py:80]: I think we agreed to use googles doc-string format

File: src/ecalc/libraries/libecalc/common/tests/integration/snapshots/test_consumer_system_v2/test_compressor_system_v2_results/consumer_system_v2.json [hunk 2]
```diff
@@ -3789,6 +3789,63 @@
...
+                        1.0
+                    ],
+                    "timesteps": [
+                        "2022-01-01 00:00:00",
+                        "2026-01-01 00:00:00"
+                    ],
+                    "typ": "STREAM_DAY",
+                    "unit": "NOK/d",
+                    "values": [
+                        0.0,
```
Comment (jorgenengelsen) [inline src/ecalc/libraries/libecalc/common/tests/integration/snapshots/test_consumer_system_v2/test_compressor_system_v2_results/consumer_system_v2.json:3808]: 0 co2?

**Review by jorgenengelsen — APPROVED** [review]
LGTM!

File: src/ecalc/libraries/libecalc/common/tests/integration/snapshots/test_consumer_system_v2/test_compressor_system_v2_results/consumer_system_v2.json (hunk 2 above)
Comment (jsolaas) [reply src/ecalc/libraries/libecalc/common/tests/integration/snapshots/test_consumer_system_v2/test_compressor_system_v2_results/consumer_system_v2.json:3808]: 0 quota for co2, 85.8 tons co2 (rate below)

**Review by TeeeJay — APPROVED** [review]

**Pull request merged**